x-pack/metricbeat/module/gcp/timeseries_metadata_collector.go

```diff
@@ -100,6 +100,13 @@ func (s *StackdriverTimeSeriesMetadataCollector) Metadata(ctx context.Context, i
 		// common.Mapstr seems to not work as expected when deleting keys so I have to iterate over all results to add
 		// the ones I want
 		for k, v := range s.timeSeries.Resource.Labels {
+
+			// We are omitting some labels here because they are added separately for services with additional metadata logic.
+			// However, we explicitly include the instance_id label to ensure it is not missed for services without additional metadata logic.
+			if k == TimeSeriesResponsePathForECSInstanceID {
+				_, _ = ecs.Put(ECSCloudInstanceIDKey, v)
+			}
+
 			if k == TimeSeriesResponsePathForECSAvailabilityZone || k == TimeSeriesResponsePathForECSInstanceID || k == TimeSeriesResponsePathForECSAccountID {
```

Comment:
We can remove this check of "TimeSeriesResponsePathForECSInstanceID" from here since we are already taking action for it in the above line?